src/utils/gameStateUtil.ts

```diff
@@ -27,13 +27,15 @@ export const gameStateUtil = {
     return gameState;
   },
   save: (state: unknown): void => {
-    if (isGameState(state)) {
-      databaseUtil.save(state);
-      return;
+    if (!isGameState(state)) {
+      throw new DeveloperError(
+        'Attempted to save an invalid game state. Did you forget to compile the game state schema?',
+      );
     }
 
-    throw new DeveloperError(
-      'Attempted to save an invalid game state. Did you forget to compile the game state schema?',
-    );
+    const isSaved = databaseUtil.save(state);
+    if (!isSaved) {
+      throw new DeveloperError('Database failed to save the game state.');
```

Comment:
DeveloperError is meant to be for errors that are unreachable in production, but this could happen if the game is run and there is a problem with the file system. I made [this ticket](https://github.com/normal-devs/klicker-knight/projects/1#card-77225594) for re-evaluating DeveloperError errors and handling them in the ✨ future ✨

Reply:
Fyi: That link is behaving weird for me. I have to click it and then refresh the page for the ticket to load and get highlighted

Reply:
The link works fine for me.